awsutil/client_options.go

```diff
@@ -98,60 +87,53 @@ func (o *ClientOptions) Validate() error {
 	return nil
 }
 
-// GetCredentials retrieves the appropriate credentials to use for the client.
-func (o *ClientOptions) GetCredentials() (*credentials.Credentials, error) {
-	if o.Role == nil && o.Creds == nil {
```

Comment:
[LoadDefaultConfig](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#loading-aws-shared-configuration) iterates through a chain of sources looking for credentials. Since one of them was looking in the environment it seemed that it was no longer mandatory to provide them explicitly since they could be in the environment.

We could still require they be provided explicitly, though it'll make it more challenging to use the environment creds as we're using them for tests.

Reply:
Eh, I think it's alright if it's the default. This change SGTM.